tools/ncf_constraints.py

```diff
@@ -0,0 +1,90 @@
+import re
+import subprocess
+import json
+import os.path
+import shutil
+import sys
+import os
+import codecs
+
+
+### Constraint checking function 
+
+def from_list( parameter, accepted_result):
+  return parameter in accepted_result
+
+def max_length( parameter, max_size):
+  return parameter.length <= max_size
+
+def min_length( parameter, min_size):
+  return parameter.length >= min_size
+
+def match_regexp(parameter, regexp):
+  match = re.match(regexp, parameter, re.S)
```

Comment:
Also, this should be ```re.search``` instead, see https://docs.python.org/2/library/re.html#search-vs-match (re.match only matches at the beginning of a string)